src/components/ReportingConfig/ReportingConfigForm.jsx

```diff
@@ -0,0 +1,512 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import isEmpty from 'lodash/isEmpty';
+import { ValidationFormGroup, Input, StatefulButton, Icon, Button } from '@edx/paragon';
+
+//  All the fields in this form that need to be validated can be added here.
+const REQUIRED_FIELDS = [
+  'hourOfDay',
+];
+// for the email delivery method
+const REQUIRED_EMAIL_FIELDS = [
+  ...REQUIRED_FIELDS,
+  'emailRaw',
+];
+// for the sftp delivery mothod
+const REQUIRED_SFTP_FIELDS = [
+  ...REQUIRED_FIELDS,
+  'sftpPort',
+  'sftpHostname',
+  'sftpUsername',
+  'sftpPassword',
+  'sftpFilePath',
+];
+
+// for the submit button
+const SUBMIT_STATES = {
+  ERROR: 'error',
+  DEFAULT: 'default',
+  COMPLETE: 'complete',
+  PENDING: 'pending',
+};
+
+const MONTHLY_MAX = 31;
+const MONTHLY_MIN = 1;
+
+const EMAIL_REGEX = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/igm;
+
+
+class ReportingConfigForm extends React.Component {
+  state = {
+    frequency: this.props.config ? this.props.config.frequency : 'monthly',
+    deliveryMethod: this.props.config ? this.props.config.deliveryMethod : 'email',
+    invalidFields: {},
+    active: this.props.config ? this.props.config.active : false,
+    submitState: SUBMIT_STATES.DEFAULT,
+  }
+
+  /**
+   * Validates this form. If the form is valid, it will return true.
+   * Otherwise, it returns the invalid fields.
```

Comment:
This seems a bit confusing once you get into the code and have lines like: `if (invalidFields !== true)`. I think it'd make more sense if it always returned the invalid fields, and then the lack of invalid fields is your check that it's confirmed. 

As a general rule, having inconsistent return types can be confusing.

Reply:
I've changed this. Now it will just return an empty object. In turn the if statement makes more sense than before. `if (!isEmpty(invalidFields))` feels better.